cmd/localizer/localizer.go

```diff
@@ -111,6 +111,19 @@ func main() { //nolint:funlen,gocyclo
 						return errors.Wrap(err, "failed to resolve service ports")
 					}
 
+					// if we couldn't find endpoints, then we fall back to binding whatever the
```

Comment:
Took the time to fix the named port handling logic to only occur when;

 * no endpoints exist on a service, so we can't resolve the ports
 * a TargetPort is, in fact, a string

When this happens, we now also warn to let the user know that this behaviour is unexpected.